### ISSUE TITLE
How to predict digit from my image

### ISSUE THREAD

**Issue by Aleksey1555** [SciSharp/TensorFlow.NET#558]
Hello every one!
I am new in tensorflow.net)
I want to create program which will be predict handwritten digits using MNIST, but I want to use my image to predict.
Here is my code:
```
Datasets<MnistDataSet> data;
data = MnistModelLoader.LoadAsync(@"C:\Users\Артём\Desktop\", oneHot: true, showProgressInConsole: false).Result;
var x = tf.placeholder(tf.float32, new TensorShape(-1, 784 ),"x");
var y_ = tf.placeholder(tf.float32, new TensorShape(-1, 10 ),"y_");
var W = tf.Variable(tf.zeros(new TensorShape(784,10)));
var b = tf.Variable(tf.zeros(new TensorShape(10)));            
Tensor y = tf.nn.softmax(tf.matmul(x, W) + b);

Tensor cross_entropy = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(labels: y_, logits: y));
Operation train_step = tf.train.GradientDescentOptimizer(0.5f).minimize(cross_entropy);

Operation init = tf.global_variables_initializer();
Session s = tf.Session();
s.run(init);

int d = 0;
for (d = 0; d < 2000; d++)
{ 
   (var xs, var ys) = data.Train.GetNextBatch(100);
   s.run(train_step, (x,xs),(y_,ys));
}

Tensor correctpredict=tf.equal(tf.argmax(y, 1), tf.argmax(y_, 1));
Tensor accur= tf.reduce_mean(tf.cast(correctpredict, tf.float32));
NDArray test_accur=s.run(accur, (x,data.Test.Data),(y_, data.Test.Labels));
Console.WriteLine("accuracy: " + (((float)test_accur) * 100.0f).ToString());

// Loading image
NDArray img = ReadTensorFromImageFile(@"ForTest.jpg", 28, 28);

Graph graph = sess.graph.as_default();

// I don't understand what string specify here "x" or something else
Tensor in1= graph.OperationByName("x").output;
// and what string here
Tensor out1 = graph.OperationByName("Softmax").output;
            
NDArray symbols=np.array(new[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 });

// here's the place that I am no sure
// and here I get error : Tensorflow.TensorflowException: 'In[0] is not a matrix. Instead it has shape [1,28,28,3]
//                                                                                 [[{{node MatMul}}]]'

var results = sess.run(out1,(in1,img)); 
results = np.squeeze(results);
int idx = np.argmax(results);
```

Function - ReadTensorFromImageFile, this is the function from here [https://tensorflownet.readthedocs.io/en/latest/ImageRecognition.html](url)

Please help!

**Comment by Oceania2018** [comment]
@Aleksey1555 Please follow the example in [branch 1.15](https://github.com/SciSharp/SciSharp-Stack-Examples/tree/tf.net-1.15).

**Comment by Aleksey1555** [comment]
Hello.
Thanks for you answer;
I try _DigitRecognitionCNN.cs_ example. It works, but i try to insert my peace of code in Test() function in the end
Here code from example:
```
 (loss_test, accuracy_test) = sess.run((loss, accuracy), (x, x_test), (y, y_test));
 print("---------------------------------------------------------");
 print($"Test loss: {loss_test.ToString("0.0000")}, test accuracy: {accuracy_test.ToString("P")}");
 print("---------------------------------------------------------");
```
And here is my peace of code:
```
NDArray img = ReadTensorFromImageFile(@"my_image.jpg");
var results=sess.run((loss, accuracy), (x, img));
```

And I get error: 
**System.Exception: 'Tensor tf.Tensor 'Train/Loss/loss:0' shape=() dtype=TF_FLOAT is not an element of this graph.'**

**Comment by Oceania2018** [comment]
@Aleksey1555 Please follow this complete [example](https://github.com/SciSharp/SciSharp-Stack-Examples/blob/master/src/TensorFlowNET.Examples/ImageProcessing/CnnInYourOwnData.cs).

**Comment by Aleksey1555** [comment]
ok, thanks
